backend/src/schema/resolvers/searches.spec.js

```diff
@@ -0,0 +1,444 @@
+          ])
+        })
+
+        describe('a post which content contains the title of the first post', () => {
+          describe('query contains the title of the first post', () => {
+            it('finds both posts', async () => {
+              variables = { query: 'beitrag' }
+              await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                data: {
+                  findResources: expect.arrayContaining([
+                    {
+                      __typename: 'Post',
+                      id: 'a-post',
+                      title: 'Beitrag',
+                      content: 'Ein erster Beitrag',
+                    },
+                    {
+                      __typename: 'Post',
+                      id: 'b-post',
+                      title: 'Aufruf',
+                      content: 'Jeder sollte seinen Beitrag leisten.',
+                    },
+                  ]),
+                },
+              })
+            })
+          })
+        })
+
+        describe('a post that contains a hyphen between two words and German quotation marks', () => {
+          describe('hyphens in query', () => {
+            it('will be treated as ordinary characters', async () => {
+              variables = { query: 'tee-ei' }
+              await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                data: {
+                  findResources: [
+                    {
+                      __typename: 'Post',
+                      id: 'g-post',
+                      title: 'Zusammengesetzte Wörter',
+                      content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                    },
+                  ],
+                },
+              })
+            })
+          })
+
+          describe('German quotation marks in query to test unicode characters (\u201E ... \u201C)', () => {
+            it('will be treated as ordinary characters', async () => {
+              variables = { query: '„teeei“' }
+              await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                data: {
+                  findResources: [
+                    {
+                      __typename: 'Post',
+                      id: 'g-post',
+                      title: 'Zusammengesetzte Wörter',
+                      content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                    },
+                  ],
+                },
+              })
+            })
+          })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-17T09:43:25Z" title="Tuesday, March 17th 2020, 10:43:25 am +01:00">Mar 17, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,452 @@
+import Factory, { cleanDatabase } from '../../db/factories'
+import { gql } from '../../helpers/jest'
+import { getNeode, getDriver } from '../../db/neo4j'
+import createServer from '../../server'
+import { createTestClient } from 'apollo-server-testing'
+
+let query, authenticatedUser
+
+const driver = getDriver()
+const neode = getNeode()
+
+beforeAll(async () => {
+  await cleanDatabase()
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  query = createTestClient(server).query
+})
+
+afterAll(async () => {
+  await cleanDatabase()
+})
+
+const searchQuery = gql`
+  query($query: String!) {
+    findResources(query: $query, limit: 5) {
+      __typename
+      ... on Post {
+        id
+        title
+        content
+      }
+      ... on User {
+        id
+        slug
+        name
+      }
+    }
+  }
+`
+let user
+
+describe('resolvers', () => {
+  describe('searches', () => {
+    let variables
+
+    describe('given one user', () => {
+      beforeAll(async () => {
+        user = await Factory.build('user', {
+          id: 'a-user',
+          name: 'John Doe',
+          slug: 'john-doe',
+        })
+        authenticatedUser = await user.toJson()
+      })
+
+      const factoryOptions = {
+        authorId: 'a-user',
+      }
+
+      describe('query contains first name of user', () => {
+        it('finds the user', async () => {
+          variables = { query: 'John' }
+          await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+            data: {
+              findResources: [
+                {
+                  id: 'a-user',
+                  name: 'John Doe',
+                  slug: 'john-doe',
+                },
+              ],
+            },
+          })
+        })
+      })
+
+      describe('adding one post', () => {
+        beforeAll(async () => {
+          await Factory.build(
+            'post',
+            {
+              id: 'a-post',
+              title: 'Beitrag',
+              content: 'Ein erster Beitrag',
+            },
+            factoryOptions,
+          )
+        })
+
+        describe('query contains title of post', () => {
+          it('finds the post', async () => {
+            variables = { query: 'beitrag' }
+            await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+              data: {
+                findResources: [
+                  {
+                    __typename: 'Post',
+                    id: 'a-post',
+                    title: 'Beitrag',
+                    content: 'Ein erster Beitrag',
+                  },
+                ],
+              },
+            })
+          })
+        })
+
+        describe('casing', () => {
+          it('does not matter', async () => {
+            variables = { query: 'BEITRAG' }
+            await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+              data: {
+                findResources: [
+                  {
+                    __typename: 'Post',
+                    id: 'a-post',
+                    title: 'Beitrag',
+                    content: 'Ein erster Beitrag',
+                  },
+                ],
+              },
+            })
+          })
+        })
+
+        describe('query consists of words not present in the corpus', () => {
+          it('returns empty search results', async () => {
+            await expect(
+              query({ query: searchQuery, variables: { query: 'Unfug' } }),
+            ).resolves.toMatchObject({ data: { findResources: [] } })
+          })
+        })
+
+        describe('testing different post content', () => {
+          beforeAll(async () => {
+            return Promise.all([
+              Factory.build(
+                'post',
+                {
+                  id: 'b-post',
+                  title: 'Aufruf',
+                  content: 'Jeder sollte seinen Beitrag leisten.',
+                },
+                factoryOptions,
+              ),
+              Factory.build(
+                'post',
+                {
+                  id: 'g-post',
+                  title: 'Zusammengesetzte Wörter',
+                  content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                },
+                factoryOptions,
+              ),
+              Factory.build(
+                'post',
+                {
+                  id: 'c-post',
+                  title: 'Die binomischen Formeln',
+                  content: `1. binomische Formel: (a + b)² = a² + 2ab + b²
+2. binomische Formel: (a - b)² = a² - 2ab + b²
+3. binomische Formel: (a + b)(a - b) = a² - b²`,
+                },
+                factoryOptions,
+              ),
+              Factory.build(
+                'post',
+                {
+                  id: 'd-post',
+                  title: 'Der Panther',
+                  content: `Sein Blick ist vom Vorübergehn der Stäbe
+so müd geworden, daß er nichts mehr hält.
+Ihm ist, als ob es tausend Stäbe gäbe
+und hinter tausend Stäben keine Welt.`,
+                },
+                factoryOptions,
+              ),
+            ])
+          })
+
+          describe('a post which content contains the title of the first post', () => {
+            describe('query contains the title of the first post', () => {
+              it('finds both posts', async () => {
+                variables = { query: 'beitrag' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: expect.arrayContaining([
+                      {
+                        __typename: 'Post',
+                        id: 'a-post',
+                        title: 'Beitrag',
+                        content: 'Ein erster Beitrag',
+                      },
+                      {
+                        __typename: 'Post',
+                        id: 'b-post',
+                        title: 'Aufruf',
+                        content: 'Jeder sollte seinen Beitrag leisten.',
+                      },
+                    ]),
+                  },
+                })
+              })
+            })
+          })
+
+          describe('a post that contains a hyphen between two words and German quotation marks', () => {
+            describe('hyphens in query', () => {
+              it('will be treated as ordinary characters', async () => {
+                variables = { query: 'tee-ei' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: [
+                      {
+                        __typename: 'Post',
+                        id: 'g-post',
+                        title: 'Zusammengesetzte Wörter',
+                        content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                      },
+                    ],
+                  },
+                })
+              })
+            })
+
+            describe('German quotation marks in query', () => {
+              it('will be treated as ordinary characters', async () => {
+                variables = { query: '„teeei“' }
+                await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+                  data: {
+                    findResources: [
+                      {
+                        __typename: 'Post',
+                        id: 'g-post',
+                        title: 'Zusammengesetzte Wörter',
+                        content: `Ein Bindestrich kann zwischen zwei Substantiven auch dann gesetzt werden, wenn drei gleichlautende Buchstaben aufeinandertreffen. Das ist etwa bei einem „Teeei“ der Fall, das so korrekt geschrieben ist. Möglich ist hier auch die Schreibweise mit Bindestrich: Tee-Ei.`,
+                      },
+                    ],
+                  },
+                })
+              })
+            })
+          })
```


I'm confused by these two test cases... is there some docs you can link to that explain why this was an issue in the past?
I went to the link you provided https://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#Escaping_Special_Characters , but I'm not sure where the German quotation marks fit into the equation.

Also looking at the tests, it's not clear. The first test searches for `tee-ei`, which the content does contain, so I would expect it to appear in the list, but maybe it wasn't because of the German quotations, somehow?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-17T11:04:40Z" title="Tuesday, March 17th 2020, 12:04:40 pm +01:00">Mar 17, 2020</time>_

---


The quotation marks are a testcase for unicode chars. I will put that in the description.